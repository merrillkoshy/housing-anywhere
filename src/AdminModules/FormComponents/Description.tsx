import React, { useState } from "react";
import { Dropdown, Form } from "react-bootstrap";

const Description = () => {
	const [type, setType] = useState(0);
	const [asType, selectedType] = useState("--");
	const [kind, setKind] = useState(0);
	const [asKind, selectKind] = useState("--");
	const [description, setDescription] = useState("");
	const [deposit, setDeposit] = useState(0);
	const [estimatedBills, setEstimatedBills] = useState(0);
	const [minimumStayMonths, setMinimumStayMonths] = useState("");
	const [maxBookableDays, setMaxBookableDays] = useState(0);
	const [moveInWindow, setMoveInWindow] = useState(0);
	const [currentOccupancy, setCurrentOccupancy] = useState(0);
	return (
		<>
			<Form.Group className="mb-3">
				<Form.Label>Type</Form.Label>
				<Dropdown>
					<Dropdown.Toggle variant="primary" id="type">
						{asType}
					</Dropdown.Toggle>

					<Dropdown.Menu>
						<Dropdown.Item
							onClick={(e) => {
								e.preventDefault();
								setType(1);
								selectedType("House");
							}}
						>
							House
						</Dropdown.Item>
						<Dropdown.Item
							onClick={(e) => {
								e.preventDefault();
								setType(2);
								selectedType("Building");
							}}
						>
							Building
						</Dropdown.Item>
						<Dropdown.Item
							onClick={(e) => {
								e.preventDefault();
								setType(3);
								selectedType("Apartment");
							}}
						>
							Apartment
						</Dropdown.Item>
					</Dropdown.Menu>
				</Dropdown>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Kind</Form.Label>
				<Dropdown>
					<Dropdown.Toggle variant="primary" id="kind">
						{asKind}
					</Dropdown.Toggle>

					<Dropdown.Menu>
						<Dropdown.Item
							type="text"
							onClick={(e) => {
								e.preventDefault();
								setKind(1);
								selectKind("EntirePlace");
							}}
						>
							Entire Place
						</Dropdown.Item>
						<Dropdown.Item
							type="text"
							onClick={(e) => {
								e.preventDefault();
								setKind(2);
								selectKind("PrivateRoom");
							}}
						>
							Private Room
						</Dropdown.Item>
						<Dropdown.Item
							type="text"
							onClick={(e) => {
								e.preventDefault();
								setKind(3);
								selectKind("SharedRoom");
							}}
						>
							Shared Room
						</Dropdown.Item>
					</Dropdown.Menu>
				</Dropdown>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Description</Form.Label>
				<Form.Control
					type="text"
					id="description"
					value={description}
					required
					onChange={(e) => {
						e.preventDefault();
						setDescription(e.currentTarget.value);
					}}
					placeholder="Enter description"
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Deposit</Form.Label>
				<Form.Control
					type="number"
					placeholder="Enter deposit"
					value={deposit}
					id="deposit"
					onChange={(e) => {
						e.preventDefault();
						setDeposit((e.currentTarget as HTMLInputElement).valueAsNumber);
					}}
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Estimated Bills</Form.Label>
				<Form.Control
					type="number"
					placeholder="Enter estimatedBills"
					value={estimatedBills}
					id="estimatedBills"
					onChange={(e) => {
						e.preventDefault();
						setEstimatedBills(
							(e.currentTarget as HTMLInputElement).valueAsNumber
						);
					}}
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Minimum Stay Months</Form.Label>
				<Form.Control
					type="text"
					id="minimumStayMonths"
					value={minimumStayMonths}
					onChange={(e) => {
						e.preventDefault();
						setMinimumStayMonths(e.currentTarget.value);
					}}
					placeholder="Enter Minimum Stay Months"
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Max Bookable Days</Form.Label>
				<Form.Control
					type="number"
					placeholder="Enter Max Bookable Days"
					value={maxBookableDays}
					id="maxBookableDays"
					onChange={(e) => {
						e.preventDefault();
						setMaxBookableDays(
							(e.currentTarget as HTMLInputElement).valueAsNumber
						);
					}}
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Move In Window</Form.Label>
				<Form.Control
					type="number"
					placeholder="Enter Move In Window"
					value={moveInWindow}
					id="moveInWindow"
					onChange={(e) => {
						e.preventDefault();
						setMoveInWindow(
							(e.currentTarget as HTMLInputElement).valueAsNumber
						);
					}}
				/>
			</Form.Group>
			<Form.Group className="mb-3">
				<Form.Label>Current Occupancy</Form.Label>
				<Form.Control
					type="number"
					placeholder="Enter Current Occupancy"
					value={currentOccupancy}
					id="currentOccupancy"
					onChange={(e) => {
						e.preventDefault();
						setCurrentOccupancy(
							(e.currentTarget as HTMLInputElement).valueAsNumber
						);
					}}
				/>
			</Form.Group>
		</>
	);
};
export default Description;
