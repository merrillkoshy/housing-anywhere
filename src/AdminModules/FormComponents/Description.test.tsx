import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom/extend-expect";
import Description from "./Description";

describe("Description module", () => {
	const DescriptionModule = shallow(<Description />);
	it("renders", () => {
		expect(DescriptionModule).toMatchSnapshot();
	});

	it("finds a type input", () => {
		expect(DescriptionModule.find("#type").exists()).toBe(true);
	});
	it("finds a kind input", () => {
		expect(DescriptionModule.find("#kind").exists()).toBe(true);
	});
	it("finds a description input which is a string", () => {
		expect(DescriptionModule.find("#description").prop("type")).toEqual("text");
	});
	it("finds an deposit input which is a number", () => {
		expect(DescriptionModule.find("#deposit").prop("type")).toEqual("number");
	});
	it("finds an estimatedBills input which is a number", () => {
		expect(DescriptionModule.find("#estimatedBills").prop("type")).toEqual(
			"number"
		);
	});
	it("finds an minimumStayMonths input which is a string", () => {
		expect(DescriptionModule.find("#minimumStayMonths").prop("type")).toEqual(
			"text"
		);
	});
	it("finds an maxBookableDays input which is a number", () => {
		expect(DescriptionModule.find("#maxBookableDays").prop("type")).toEqual(
			"number"
		);
	});
	it("finds an moveInWindow input which is a number", () => {
		expect(DescriptionModule.find("#moveInWindow").prop("type")).toEqual(
			"number"
		);
	});
	it("finds an currentOccupancy input which is a number", () => {
		expect(DescriptionModule.find("#currentOccupancy").prop("type")).toEqual(
			"number"
		);
	});
});
