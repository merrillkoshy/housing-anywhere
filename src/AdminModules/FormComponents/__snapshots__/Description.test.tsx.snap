// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description module renders 1`] = `
<Fragment>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Type
    </FormLabel>
    <Dropdown
      navbar={false}
    >
      <DropdownToggle
        id="type"
        variant="primary"
      >
        --
      </DropdownToggle>
      <DropdownMenu
        align="left"
        alignRight={false}
        flip={true}
      >
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
        >
          House
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
        >
          Building
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
        >
          Apartment
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Kind
    </FormLabel>
    <Dropdown
      navbar={false}
    >
      <DropdownToggle
        id="kind"
        variant="primary"
      >
        --
      </DropdownToggle>
      <DropdownMenu
        align="left"
        alignRight={false}
        flip={true}
      >
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
          type="text"
        >
          Entire Place
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
          type="text"
        >
          Private Room
        </DropdownItem>
        <DropdownItem
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            }
          }
          disabled={false}
          onClick={[Function]}
          type="text"
        >
          Shared Room
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Description
    </FormLabel>
    <FormControl
      id="description"
      onChange={[Function]}
      placeholder="Enter description"
      required={true}
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Deposit
    </FormLabel>
    <FormControl
      id="deposit"
      onChange={[Function]}
      placeholder="Enter deposit"
      type="number"
      value={0}
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Estimated Bills
    </FormLabel>
    <FormControl
      id="estimatedBills"
      onChange={[Function]}
      placeholder="Enter estimatedBills"
      type="number"
      value={0}
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Minimum Stay Months
    </FormLabel>
    <FormControl
      id="minimumStayMonths"
      onChange={[Function]}
      placeholder="Enter Minimum Stay Months"
      type="text"
      value=""
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Max Bookable Days
    </FormLabel>
    <FormControl
      id="maxBookableDays"
      onChange={[Function]}
      placeholder="Enter Max Bookable Days"
      type="number"
      value={0}
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Move In Window
    </FormLabel>
    <FormControl
      id="moveInWindow"
      onChange={[Function]}
      placeholder="Enter Move In Window"
      type="number"
      value={0}
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Current Occupancy
    </FormLabel>
    <FormControl
      id="currentOccupancy"
      onChange={[Function]}
      placeholder="Enter Current Occupancy"
      type="number"
      value={0}
    />
  </FormGroup>
</Fragment>
`;
